### PR TITLE
Improved docs for installation

### PR DESCRIPTION
**Pull request by twentyse7en**
added steps for installing LiberTEM for code contribution.

## Contributor Checklist:

* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -135,20 +135,36 @@ continue by reading the :ref:`usage documentation`.
...
+
+#. Log into your `GitHub <https://github.com/>`_ account.
+
+#. Go to the `LiberTEM GitHub <https://github.com/liberteM/LiberTEM/>`_ home page.
+
+#. Click on the *fork* button:
+
+    ..  figure:: ./images/forking_button.png
+
+#. Copy your fork of LiberTEM from GitHub to your computer
```
Comment (uellue) [inline docs/source/install.rst]: Better "clone" instead of "copy"?

File: docs/source/install.rst [hunk 2]
```diff
@@ -135,20 +135,36 @@ continue by reading the :ref:`usage documentation`.
...
+Or if you wish to contribute to LiberTEM, follow these steps instead :
+
+#. Log into your `GitHub <https://github.com/>`_ account.
+
+#. Go to the `LiberTEM GitHub <https://github.com/liberteM/LiberTEM/>`_ home page.
+
+#. Click on the *fork* button:
+
+    ..  figure:: ./images/forking_button.png
+
```
Comment (uellue) [inline docs/source/install.rst]: Perhaps better to link to GitHub's documentation on the topic? https://help.github.com/en/github/getting-started-with-github/fork-a-repo

**Comment by uellue** [comment]
Looking good, in general!

I am unsure if one should just refer to the GitHub documentation and remove the screenshot and detailed steps, to make sure that new users read a bit into it instead of just blindly following instructions, or if blindly following the steps is what we want to put new contributors on the right path so that they don't get lost or discouraged in the big GitHub documentation.

@twentyse7en, how and where did you learn to do a good pull request since you nailed it on the first try? Was it a simple recipe, did you read into background documentation, or did someone show you or coach you?  We should perhaps point new contributors in the direction where you learned it. :-)

#603 should fix the changelog issue, by the way.

**Comment by twentyse7en** [comment]
> Looking good, in general!
> 
> I am unsure if one should just refer to the GitHub documentation and remove the screenshot and detailed steps, to make sure that new users read a bit into it instead of just blindly following instructions, or if blindly following the steps is what we want to put new contributors on the right path so that they don't get lost or discouraged in the big GitHub documentation.
> 
> @twentyse7en, how and where did you learn to do a good pull request since you nailed it on the first try? Was it a simple recipe, did you read into background documentation, or did someone show you or coach you? We should perhaps point new contributors in the direction where you learned it. :-)
> 
> #603 should fix the changelog issue, by the way.

First step I did was taking this course [version control with git](https://www.udacity.com/course/version-control-with-git--ud123). This helped a lot to understand the basics of Github.

Did a demo PR with [first-contribution](https://github.com/firstcontributions/first-contributions). This really helps to gain some confidence.

Previously I have joined SunPy organisation, they recommended to go through [workflow](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html)

**Comment by twentyse7en** [comment]
I blindly followed Installation guide in LiberTEM and got a error when I push the changes. Beginners should go through a workflow guide.

**Comment by uellue** [comment]
> First step I did was taking this course [version control with git](https://www.udacity.com/course/version-control-with-git--ud123). This helped a lot to understand the basics of Github.
> 
> Did a demo PR with [first-contribution](https://github.com/firstcontributions/first-contributions). This really helps to gain some confidence.
> 
> Previously I have joined SunPy organisation, they recommended to go through [workflow](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html)

Hi @twentyse7en, all these are great resources! Perhaps you could include them here and suggest that contributors who are new to git and/or GitHub check them out? All three together make a great combination and evidently put you on the right path. :-)

The CI issue is resolved as well with merging #603, by the way.

**Comment by twentyse7en** [comment]
> > First step I did was taking this course [version control with git](https://www.udacity.com/course/version-control-with-git--ud123). This helped a lot to understand the basics of Github.
> > Did a demo PR with [first-contribution](https://github.com/firstcontributions/first-contributions). This really helps to gain some confidence.
> > Previously I have joined SunPy organisation, they recommended to go through [workflow](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html)
> 
> Hi @twentyse7en, all these are great resources! Perhaps you could include them here and suggest that contributors who are new to git and/or GitHub check them out? All three together make a great combination and evidently put you on the right path. :-)
> 
> The CI issue is resolved as well with merging #603, by the way.

Should I add the contents as a part of "For more Information"

**Comment by uellue** [comment]
> Should I add the contents as a part of "For more Information" ?

Yes, sounds good! 👍

File: docs/source/install.rst [hunk 3]
```diff
@@ -160,6 +160,15 @@ Or if you wish to contribute to LiberTEM, follow these steps instead :
 
 For more information about `forking a repository
 <https://help.github.com/en/github/getting-started-with-github/fork-a-repo>`_.
+If you have limited experience using git, consider going through following
```
Comment (uellue) [inline docs/source/install.rst]: Maybe "For a beginner-friendly introduction to git and GitHub, ..."? That's more from the perspective of what new users might want.

Also "going through the following", note the "the". :-)

File: docs/source/install.rst [hunk 4]
```diff
@@ -160,6 +160,15 @@ Or if you wish to contribute to LiberTEM, follow these steps instead :
...
+resources:
+
+* This `free course <https://www.udacity.com/course/version-control-with-git--ud123>`_
+  covers the essentials of using Git.
+* For practicing `pull request <https://github.com/firstcontributions/first-contributions>`_
```
Comment (uellue) [inline docs/source/install.rst]: "Practice pull requests in a safe sandbox environment", perhaps? Addressing the underlying fear of new contributors more directly.

File: docs/source/install.rst [hunk 5]
```diff
@@ -160,6 +160,15 @@ Or if you wish to contribute to LiberTEM, follow these steps instead :
...
+  for first time.
+* Sample `Workflow <https://docs.astropy.org/en/latest/development/workflow/development_workflow.html>`_
```
Comment (uellue) [inline docs/source/install.rst]: Capitalization, lower-case "workflow"

**Comment by twentyse7en** [comment]
Thx for pointing out. I think I'm terrible at my documentation skills :sweat_smile:. Will update the docs as you recommend.

**Comment by twentyse7en** [comment]
We should also create a workflow similar to astropy mentioned above. Also [worked example](http://docs.astropy.org/en/stable/development/workflow/git_edit_workflow_examples.html) would be great deal for beginners. Doing PR for first time was really hard for me. Also encourage to ask questions, beginners hesitate to ask.

**Comment by sk1p** [comment]
> I think I'm terrible at my documentation skills sweat_smile

Don't feel bad, writing good documentation is **hard**!

> We should also create a workflow similar to astropy mentioned above.
> Also worked example would be great deal for beginners.

Agreed! Would be nice as an additional document.

> Doing PR for first time was really hard for me.

Which part would you say was the biggest hurdle for you?

> Also encourage to ask questions, beginners hesitate to ask.

:+1: - contributing docs should mention the gitter chat, and also the possibility to ask questions via github issues.

**Comment by twentyse7en** [comment]
I made a pull request for a issue, then made further improvements. I was confused how to attach new commit with pull request I already made. Later I got it,  a simple `git push` is enough.

**Comment by uellue** [comment]
> Thx for pointing out. I think I'm terrible at my documentation skills 😅. Updated the docs as you recommend.

It was good already in the first place, don't worry! The feedback is not meant as personal criticism, we just work together to get better results than each could get on their own.

**Comment by uellue** [comment]
@twentyse7en It looks like you should rebase this pull request to the latest version of the LiberTEM master branch? After that we can merge, I think.

If you have ideas for further improvements, we could do that in a new PR.

**Comment by twentyse7en** [comment]
I have never done a `git rebase`. Is what I done right ?

**Comment by uellue** [comment]
Hm, something seems to have gone a bit sideways with the rebase. No worries, that can be tricky with git.

For the next rebase we can perhaps coordinate through Gitter and work step by step. :-) @sk1p is the expert for this.

For now, I'll try to squash and merge. Fingers crossed that it works! :-D

**Comment by twentyse7en** [comment]
What I did was 
 `git checkout branch`
`git rebase master`

**Comment by uellue** [comment]
Hm, that sounds about correct. One would have to check that the master branch used for the rebase is in sync with the upstream master. I am using `git pull --ff-only upstream master` since that will complain if the local master has diverged from upstream.

**Comment by uellue** [comment]
Anyway, we try again with the next rebase. :-)

**Pull request merged**